File: system-test/sec-tests/python/sec_59_soft_delete.py
# SEC-59: Incomplete Data Deletion / Residual Data
# Test: TC-CR-062
from models import User, db

def delete_user(user_id):
    user = User.query.get(user_id)
    user.is_deleted = True
    db.session.commit()
    # Associated orders, comments, logs not cleaned up

File: system-test/sec-tests/java/Sec59SoftDelete.java
// SEC-59: Incomplete Data Deletion / Residual Data
// Test: TC-CR-062
package com.example;

import javax.persistence.*;

@Entity
public class User {
    @Id private Long id;
    private String username;
    private boolean isDeleted = false;  // Soft delete without cascading

    public void softDelete() {
        this.isDeleted = true;
        // Associated orders, comments, audit records NOT cleaned up
    }
}
